### PR TITLE
feat: remove deprecated functions

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

File: api/ocm/selectors/json.go [hunk 1]
```diff
@@ -223,3 +224,14 @@ func (j JSONSchemaSelector) Match(obj map[string]string) (bool, error) {
 	}
 	return res.Valid(), nil
 }
+
+// TODO 1: this might not work.
+// TODO 2: what about accessor2?
```
Comment (hilmarf) [inline api/ocm/selectors/json.go:229]: are you done with this already? The rest LGTM

Comment (Skarlso) [reply api/ocm/selectors/json.go:229]: No, I need confirmation that this makes sense there. I have no idea how this works. :D

Comment (Skarlso) [reply api/ocm/selectors/json.go]: Nice, this point is now "fixed". :D

**Comment by Skarlso** [comment]
Note that the Docker section still has deprecated stuff, but we aren't removing that here, because the entire thing is removed in #904

**Review by hilmarf — APPROVED** [review]
LGTM

**Pull request merged**